tools/clang/test/CodeGenSPIRV/intrinsics.get-attribute-at-vertex.s.retStruct.hlsl

```diff
@@ -0,0 +1,55 @@
+// RUN: %dxc -T ps_6_1 -E main -spirv
+
+enum VertexID { 
+    FIRST = 0,
+    SECOND = 1,
+    THIRD = 2
+};
+
+
+struct PSInput
+{
+    nointerpolation float3 color0 : COLOR0;
+    nointerpolation float3 color1 : COLOR1;
+    nointerpolation bool3 color2 : COLOR2;
+};
+
+// CHECK: OpCapability FragmentBarycentricKHR
+// CHECK: OpExtension "SPV_KHR_fragment_shader_barycentric"
+// CHECK: OpDecorate [[color0:%\w+]] Location 0
+// CHECK: OpDecorate [[color1:%\w+]] Location 3
+// CHECK: OpDecorate [[color2:%\w+]] Location 6
```

Comment:
If I change the locations, spirv-val does not complain. However, that does not mean it is valid.

Reply:
I think I miss the case you mentioned here, location in previous stage should be changed if I do reassginment here.
I'll check linkage location mapping and make a fix. 
Or, you mean, should I just keep them with original location here?

I was comparing generated code with local case which directly use an array as its input.
I was thinking if an input is changed to an array like here, we should also calculate inputs location again here. Like:

input: float a => register 0
          float b => register 1
but after expanding here, in normal case, array's location/register number should be:
          float a[3] => register 0~2
          float b[3] => register 3~5

Reply:
> Or, you mean, should I just keep them with original location here?

If it is legal, I would try to keep the original location. If it is not legal, then we need to document it well. I am not suggesting anything specific because I'm not familiar enough with the details.

Reply:
This case has been removed